### PR TITLE
:bug: Ensure supported DID before calling Rotate

### LINKED ISSUE #3379: :bug: Unhandled `InvalidDIDError` for unqualified did during Rotate
When calling `/did-rotate/{conn_id}/rotate`, using `?to_did=` some unqualified did, we get a 200 success response, but there is an unhandled exception that arises in the agent, which goes unreported to the client:

```
Traceback (most recent call last):
  File \"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/core/dispatcher.py\", line 258, in handle_v1_message
    await handler(context, responder)
  File \"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/protocols/did_rotate/v1_0/handlers/rotate_handler.py\", line 32, in handle
    if record := await did_rotate_mgr.receive_rotate(connection_record, rotate):
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/protocols/did_rotate/v1_0/manager.py\", line 131, in receive_rotate
    await self._ensure_supported_did(rotate.to_did)
  File \"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/protocols/did_rotate/v1_0/manager.py\", line 242, in _ensure_supported_did
    await resolver.resolve(self.profile, did)
  File \"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/resolver/did_resolver.py\", line 87, in resolve
    _, doc = await self._resolve(profile, did, service_accept, timeout=timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File \"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/resolver/did_resolver.py\", line 59, in _resolve
    DID.validate(did)
  File \"/home/aries/.local/lib/python3.12/site-packages/pydid/did.py\", line 95, in validate
    return cls.model_validate(did)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File \"/home/aries/.local/lib/python3.12/site-packages/pydid/did.py\", line 89, in model_validate
    raise InvalidDIDError('\"{}\" is not a valid DID'.format(did))
pydid.did.InvalidDIDError: \"S5kDeCsZwReYzQZroSDUq8\" is not a valid DID"
```

This is because downstream from the RotateHandler, `DID.validate` gets called. But this is not validated in the DIDRotateManager before the message gets submitted to the Handler.

### PR DESCRIPTION
**Pull request by ff137**
Resolves #3379 and #3381

____

I'm open to alternatives. I first thought to just call DID.validate, and raise BadRequest if that fails, but since `ensure_supported_did` is called downstream, and the client would not immediately know if the DID is unresolvable, I figured that everything may as well be validated in the route, before being sent to the RotateHandler.

That means `ensure_supported_did` will now be called twice. Perhaps the 2nd call can now be removed, since it's now just called earlier. Open to suggestions

### CODE REVIEW

**Comment by ff137** [comment]
My attempted solution results in:
```
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/protocols/did_rotate/v1_0/routes.py", line 73, in rotate
    await did_rotate_mgr.ensure_supported_did(to_did)
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/protocols/did_rotate/v1_0/manager.py", line 242, in ensure_supported_did
    await resolver.resolve(self.profile, did)
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/resolver/did_resolver.py", line 87, in resolve
    _, doc = await self._resolve(profile, did, service_accept, timeout=timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/resolver/did_resolver.py", line 63, in _resolve
    document = await asyncio.wait_for(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/resolver/base.py", line 161, in resolve
    result = await self._resolve(profile, did, service_accept)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/resolver/default/indy.py", line 185, in _resolve
    vmethod = builder.verification_method.add(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/pydid/doc/builder.py", line 50, in add
    vmethod = type_.make(id=self._did.ref(ident), controller=controller, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/pydid/resource.py", line 101, in make
    return cls(**kwargs)
           ^^^^^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/pydid/verification_method.py", line 57, in __init__
    super().__init__(**data)
  File "/home/aries/.local/lib/python3.12/site-packages/pydantic/main.py", line 214, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 1 validation error for Ed25519VerificationKey2018
public_key_base58
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
```

...

**Comment by ff137** [comment]
RotateAck is not correctly configured. Somewhere it is trying to load class "Ack", instead of RotateAck:
```
"/home/aries/.local/lib/python3.12/site-packages/acapy_agent/utils/classloader.py", line 234, in resolved
    DeferLoad._class_cache[self._cls_path] = ClassLoader.load_class(
                                             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/aries/.local/lib/python3.12/site-packages/acapy_agent/utils/classloader.py", line 125, in load_class
    raise ClassNotFoundError(
acapy_agent.utils.classloader.ClassNotFoundError: Class 'Ack' not defined in module: acapy_agent.protocols.did_rotate.v1_0.messages.ack
```

**Comment by ff137** [comment]
@jamshale @dbluhm just a headsup, it looks like the SonarCloud coverage analysis is broken. My coverage report locally shows all changes are covered, but SonarCloud gives 0%. Presumably relates to #3390

File: acapy_agent/connections/base_manager.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import logging
 from typing import Dict, List, Optional, Sequence, Text, Tuple, Union
 
+import pydantic
```
Comment (dbluhm) [inline acapy_agent/connections/base_manager.py:10]: Nitpick: pydantic isn't a direct dependency of ACA-Py. To catch validation errors raised by PyDID, we should instead catch `ValueError` so we don't put an implicit dependency of ACA-Py directly on pydantic.

**Review by dbluhm — COMMENTED** [review]
Overall looks pretty good. Quick nitpick on handling errors raised by pydid.

**Review by jamshale — COMMENTED** [review]
👍

File: acapy_agent/connections/base_manager.py (hunk 1 above)
Comment (ff137) [reply acapy_agent/connections/base_manager.py:10]: Good point! Will do

**Review by jamshale — APPROVED** [review]

**Pull request merged**